make/autoconf/flags-cflags.m4

```diff
@@ -95,8 +95,24 @@ AC_DEFUN([FLAGS_SETUP_DEBUG_SYMBOLS],
   # info flags for toolchains unless we know they work.
   # See JDK-8207057.
   ASFLAGS_DEBUG_SYMBOLS=""
+
+  # Debug prefix mapping if supported by compiler
+  DEBUG_PREFIX_CFLAGS=
+
   # Debug symbols
   if test "x$TOOLCHAIN_TYPE" = xgcc; then
+    if test "x$ALLOW_ABSOLUTE_PATHS_IN_OUTPUT" = "xfalse"; then
+      # Check if compiler supports -fdebug-prefix-map. If so, use that to make
+      # the debug symbol paths resolve to paths relative to the workspace root.
+      workspace_root_trailing_slash="${WORKSPACE_ROOT%/}/"
+      DEBUG_PREFIX_CFLAGS="-fdebug-prefix-map=${workspace_root_trailing_slash}="
+      FLAGS_COMPILER_CHECK_ARGUMENTS(ARGUMENT: [${DEBUG_PREFIX_CFLAGS}],
```

Comment:
Ideally capability checks for the compiler should be done separately for the BUILD and TARGET compiler. If cross compiling, and using different versions of GCC for the BUILD and TARGET, we may get into trouble. That is certainly a rare case, but still something to be aware of at least. Not sure if it's worth trying to fix.

Reply:
Thanks @erikj79 , yes this check is probably a bit muted here anyway, as according to reproducible-builds.org -fdebug-prefix-map is (available in all GCC versions, Clang 3.8)